miden-crypto/src/ies/keys.rs

```diff
@@ -0,0 +1,746 @@
+use alloc::vec::Vec;
+
+use rand::{CryptoRng, RngCore};
+
+use super::{
+    crypto_box::{CryptoBox, RawSealedMessage},
+    error::IntegratedEncryptionSchemeError,
+    message::{IesAlgorithm, SealedMessage},
+};
+use crate::{
+    Felt,
+    aead::{aead_rpo::AeadRpo, xchacha::XChaCha},
+    ecdh::{KeyAgreementScheme, k256::K256, x25519::X25519},
+    utils::{Deserializable, Serializable},
+};
+
+/// Instantiation of sealed box using K256 + XChaCha20Poly1305
+type K256XChaCha20Poly1305 = CryptoBox<K256, XChaCha>;
+/// Instantiation of sealed box using X25519 + XChaCha20Poly1305
+type X25519XChaCha20Poly1305 = CryptoBox<X25519, XChaCha>;
+/// Instantiation of sealed box using X25519 + AeadRPO
+type X25519AeadRpo = CryptoBox<X25519, AeadRpo>;
+
+/// Public key for sealing messages to a recipient.
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
This enum shows some boilerplate where each method requires repetitive match statements. I notice you have very similar code blocks for each variant in methods like `seal_with_associated_data`.

Have you considered using the `enum_dispatch` crate to eliminate this boilerplate? We [talked about this before](https://github.com/0xMiden/crypto/pull/475#discussion_r2269977819)

For example, you could define a `SealingKeyTrait` and then let the macro generate the dispatch code, which would reduce the repetitive match statements while maintaining zero runtime overhead. This would make adding new algorithms much simpler too.

What are your thoughts on using enum_dispatch here?

Reply:
The main potential issue with `enum_dispatch` solution is the fact that there are some variants which share the same inner type which should create problems.

Reply:
I have go on and used some helper macros in order to reduce the boilerplate code